prov/mxm/configure.m4

```diff
@@ -0,0 +1,86 @@
+dnl Configury specific to the libfabrics mxm provider
+
+dnl Called to configure this provider
+dnl
+dnl Arguments:
+dnl
+dnl $1: action if configured successfully
+dnl $2: action if not configured successfully
+dnl
+AC_DEFUN([FI_MXM_CONFIGURE],[
+	# Determine if we can support the mxm provider
+        mxm_happy=0
+        AC_ARG_ENABLE([mxm],
+		[AS_HELP_STRING([--enable-mxm],
+                [Enable MXM provider @<:@default=no@:>@])
+		],
+		[],
+                [enable_mxm=no])
+        AC_CHECK_SIZEOF([void *])
+
+        AS_IF([test x"$enable_mxm" = x"yes"],
+              [AS_IF([test x"$ac_cv_sizeof_void_p" = x"8"],
+                     [],
+                     [
+                         enable_mxm=no
+                         AC_MSG_WARN([MXM OFI provider does not support 32 bit target platform])
+                     ])],
+              [])
+
+        AC_ARG_WITH([mxm],
+                [AS_HELP_STRING([--with-mxm=@<:@MXM installation path@:>@],
+                        [Provide path to MXM installation])
+		],
+		[AS_CASE([$with_mxm],
+                        [yes|no], [AC_DEFINE([HAVE_MXM], [1], [Define if MXM is enabled])],
+                        [CPPFLAGS="-I$with_mxm/include $CPPFLAGS"
+			LDFLAGS="-L$with_mxm/lib -Wl,-rpath=$with_mxm/lib $LDFLAGS"
```

Comment:
I don't think we should set the rpath, just the library search path should be fine (-L).